jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/commands/LogCleanupCommand.java

```diff
@@ -268,6 +265,12 @@ public ExecutionResults execute(CommandContext ctx) throws Exception {
                 executionResults.setData("ProcessInstanceLogRemoved", piLogsRemoved);
             }
         }
+        // calculate nextRun at the end of the execution
+        String nextRun = (String) ctx.getData("NextRun");
+        if (nextRun != null) {
+            long executionTime = Instant.now().minus(startTime, ChronoUnit.MILLIS).toEpochMilli();
+            nextScheduleTimeAdd = DateTimeUtils.parseDateAsDuration(nextRun) - executionTime;
```

Comment:
You are modifying the behaviour for every command in here. 
I would add a new flag in here saying Repeated FIXED_RATE, OR FIXED_INTERVAL

FIXED_RATE = scheduled + interval
FIXED_INTERVAL = end of execution + interval (default behaviour)

Reply:
Added a parameter to the command, with the default behavior being inline with how it has been calculated so far. Note that I also added a base class for a central place for the calculation.